src/cpu/aarch64/jit_aarch64_sve_512_conv_kernel.cpp

```diff
@@ -0,0 +1,4417 @@
+
+    const int full_simd_w = cpu_isa_traits<sve_512>::vlen / typesize;
+    jcp.simd_w = full_simd_w;
+    jcp.oc_block = jcp.simd_w;
+    jcp.ic_block = jcp.is_1stconv ? jcp.ic : jcp.simd_w;
+
+    /* Channel padding */
+    if (ok_to_pad_channels) {
+        jcp.oc = rnd_up(jcp.oc, jcp.oc_block);
+        jcp.ic = rnd_up(jcp.ic, jcp.ic_block);
+    }
+
+    /* Input and output channels must be multiples of simd_w */
+    if (!(jcp.oc % jcp.oc_block == 0 && jcp.ic % jcp.ic_block == 0)) {
+        return status::unimplemented;
+    }
+    jcp.ic_tail = 0;
+    jcp.oc_tail = 0;
+
+    /* Post operation check */
+    if (!post_ops_ok(jcp, attr)) { return status::unimplemented; }
+
+    /* Eltwise operation check */
+    const auto &p = attr.post_ops_;
+    jcp.with_sum = p.find(primitive_kind::sum) != -1;
+    const int eltwise_ind = p.find(primitive_kind::eltwise);
+    jcp.with_eltwise = eltwise_ind != -1;
+    if (jcp.with_eltwise) {
+#ifndef DISABLE_ELTWISE
+        jcp.eltwise = p.entry_[eltwise_ind].eltwise;
+        if (jcp.eltwise.alg == alg_kind::eltwise_pow)
+            return status::unimplemented;
+        if (dst_d.data_type() == data_type::s32) return status::unimplemented;
+#else
+        return status::unimplemented;
+#endif
+    }
+
+    format_tag_t src_tag, dst_tag, wei_tag;
+
+    dst_tag = dat_tag_nCx16c;
+    src_tag = jcp.is_1stconv ? dat_tag_ncx : dat_tag_nCx16c;
+    wei_tag = pick(2 * ndims - 6 + with_groups, OIw16i16o, gOIw16i16o,
+            OIhw16i16o, gOIhw16i16o, OIdhw16i16o, gOIdhw16i16o);
+
+    if (src_md.format_kind == format_kind::any)
+        CHECK(memory_desc_init_by_tag(src_md, src_tag));
+    else if (curr_src_tag != src_tag)
+        return status::unimplemented;
+    jcp.src_tag = src_tag;
+
+    if (dst_md.format_kind == format_kind::any)
+        CHECK(memory_desc_init_by_tag(dst_md, dst_tag));
+    else if (curr_dst_tag != dst_tag)
+        return status::unimplemented;
+    jcp.dst_tag = dst_tag;
+
+    jcp.with_bias = cd.bias_desc.format_kind != format_kind::undef;
+    if (jcp.with_bias) {
+        if (bias_d.format_kind() == format_kind::any)
+            CHECK(memory_desc_init_by_tag(bias_md, x));
+    }
+
+    if (mayiuse(sve_512) && src_d.data_type() == data_type::f32
+            && weights_d.data_type() == data_type::f32
+            && dst_d.data_type() == data_type::f32) {
+        jcp.ver = ver_fma;
+        jcp.typesize_in = typesize;
+        jcp.typesize_out = typesize;
+
+        if (jcp.is_1stconv) {
+            /* NOTE:
+             * 1) When memory-protection is enabled, this guards against a
+             * seg-fault from illegal memory access. A potential solution is
+             * to enable tail processing within 'compute_loop_4fma_1st'
+             * 2) 4FMA Kernel does not support:
+             *  `l_pad > 0 || r_pad > 0`; when `kw > 1`
+             * from incorrect 'get_ow_start' and 'get_ow_end' calculation, so
+             * disable for now. */
```

Comment:
(I don't think these comments are relevant to this JIT version)

Reply:
Thanks for your reviews. I'll delete it.

Reply:
LGM